config/manager/manager.yaml

```diff
@@ -5,15 +5,18 @@ metadata:
   namespace: system
   labels:
     control-plane: controller-manager
+    webhook: controller-manager
 spec:
   selector:
     matchLabels:
       control-plane: controller-manager
+      webhook: controller-manager
```

Comment:
This is the last piece of puzzle, let me explain why. Before that, I'd like to say this is manually added and totally customizable and is not needed if BMO runs standalone.

Capm3 runs three controller-managers under `capm3-system` namespace;
- capm3-controller-manager
- capm3-ipam-controller-manager
- capm3-baremetal-operator-controller-manager

These managers are all have `control-plane: controller-manager` labels. If we enable webhook by applying this service patch https://github.com/metal3-io/baremetal-operator/blob/22484ec5ff91aeaac912cf7188e8af55fcee099a/config/webhook/service_patch.yaml#L12, webhook traffic is routed to each of these managers, because they all have matching labels. 2/3 of webhook requests are failing. Because only baremetal-operator-controller-manager serves webhook endpoint(and others throw connection refused error).

I added this to distinguish baremetal-operator-controller-manager and webhook requests only goes to this manager. The other option is changing baremetal-operator-controller-managers label(i.e. `control-plane: bmo-controller-manager`).

Or there is another way I don't know.

Reply:
According to the @asalkeld suggestion, I removed new label from selector part. Because @zaneb you are right, that might cause problems in upgrade.

Reply:
@zaneb now this PR also covers CAPM3 v1alpha4 and v1alpha5. The above change(adding new label) is only needed for v1alpha4. Since BMO should support both versions, it is still relevant.